src/handler.rs

```diff
@@ -95,7 +96,7 @@ pub async fn serverless(
     let execution_timer_start = Instant::now();
 
     // reserve cgroup
-    let cgroup = appstate.cgroups.lock().unwrap().reserve();
+    let cgroup = appstate.cgroups.lock().await.reserve();
```

Comment:
For cgroups use std::mutex, because tokio ones are costly, and not required for cgroup . For ref: https://tokio.rs/tokio/tutorial/shared-state. In model.rs, separate the usage of it.